tap_sailthru/streams.py

```diff
@@ -158,10 +158,17 @@ def prepare_request_payload(
         :param context: Stream partition or context dictionary.
         :returns: dict, A dictionary containing the request payload.
         """
+
+        # set the start date to 7 days before the last replication state
+        starting_replication_time = self.stream_state.get("starting_replication_value")
+        starting_replication_date = pendulum.parse(starting_replication_time)
+        starting_replication_date_minus_7 = starting_replication_date.subtract(days=7).start_of('day')
+        start_date = starting_replication_date_minus_7.to_date_string()
```

Comment:
what happens if the `starting_replication_value` is null? or is it never null?

Reply:
oop good point, let me make a change to take the `start_date` from the config instead then